astroquery/desi/core.py

```diff
@@ -0,0 +1,140 @@
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates`.
+            coordinates around which to query
+        radius : str or `astropy.units.Quantity`.
+            the radius of the cone search
+
+        Returns
+        -------
+        response : `requests.Response`
+            The HTTP response returned from the service.
+            All async methods should return the raw HTTP response.
+        """
+
+        url = 'https://datalab.noirlab.edu/tap'
+        tap_service = vo.dal.TAPService(url)
+        qstr = "SELECT all * FROM ls_dr" + str(data_release) + ".tractor WHERE dec>" + str(coordinates.dec.deg - radius.deg) + " and dec<" + str(
+            coordinates.dec.deg + radius.deg) + " and ra>" + str(coordinates.ra.deg - radius.deg / np.cos(coordinates.dec.deg * np.pi / 180.)) + " and ra<" + str(
+            coordinates.ra.deg + radius.deg / np.cos(coordinates.dec.deg * np.pi / 180))
+
+        tap_result = tap_service.run_sync(qstr)
+        tap_result = tap_result.to_table()
+        # filter out duplicated lines from the table
+        mask = tap_result['type'] != 'D'
+        filtered_table = tap_result[mask]
+
+        return filtered_table
+
+    def get_images_async(self, position, survey, coordinates=None, data_release=9,
+                         projection=None, pixels=None, scaling=None,
+                         sampler=None, resolver=None, deedger=None, lut=None,
+                         grid=None, gridlabels=None, radius=None, height=None,
+                         width=None, cache=True, show_progress=True, image_band='g'):
+        """
+        Returns
+        -------
+        A list of context-managers that yield readable file-like objects
+        """
+
+        image_size_arcsec = radius.arcsec
+        pixsize = 2 * image_size_arcsec / pixels
+
+        image_url = 'https://www.legacysurvey.org/viewer/fits-cutout?ra=' + str(position.ra.deg) + '&dec=' + str(position.dec.deg) + '&size=' + str(
+            pixels) + '&layer=ls-dr' + str(data_release) + '&pixscale=' + str(pixsize) + '&bands=' + image_band
+
+        print("image_url: ", image_url)
+
+        file_container = commons.FileContainer(image_url, encoding='binary', show_progress=show_progress)
+
+        try:
+            fits_file = file_container.get_fits()
+        except (requests.exceptions.HTTPError, urllib.error.HTTPError) as e:
+            # TODO not sure this is the most suitable exception
+            raise NoResultsWarning(f"{str(e)} - Problem retrieving the file at the url: {str(image_url)}")
+
+        return [fits_file]
+
+    def _parse_result(self, responses, verbose=False):
+        tables_list = []
+        output_table = Table()
+        tmp_dir_list = []
+
+        # handles the cases of query_region with TAP which returns a Table
+        # or get_images which should return a list of HDUList, and here in DESILegacySurvey
+        # a list of one HDUList object is returned
```

Comment:
What are the cases where this is not true?

Reply:
Thanks for the comment, that is a valid point I was not considering. Answer is quite simple: no it does not at the current state. So, it can be removed.

Reply:
It has now been removed.